packages/mobx-react/README.md

```diff
@@ -87,6 +88,90 @@ class TodoView extends React.Component {
 const TodoView = observer(({ todo }) => <div>{todo.title}</div>)
 ```
 
+##### Note on using props and state in derivations
+
+`mobx-react` version 6 and lower would automatically turn `this.state` and `this.props` into observables.
+This has the benefit that computed properties and reactions were able to observe those.
+However, since this pattern is fundamentally incompatible with `StrictMode` in React 18.2 and higher, this behavior has been removed in React 18.
+
+As a result, we recommend to no longer mark properties as `@computed` in observer components if they depend on `this.state` or `this.props`.
+
+```javascript
+@observer
+class Doubler extends React.Component<{ counter: number }> {
+    @computed // BROKEN! <-- @computed should be removed in mobx-react > 7
+    get doubleValue() {
+        // Changes to this.props will no longer be detected properly, to fix it,
+        // remove the @computed annotation.
+        return this.props * 2
+    }
+
+    render() {
+        return <div>{this.doubleValue}</div>
+    }
+}
+```
+
+Similarly, reactions will no longer respond to `this.state` / `this.props`. This can be overcome by creating an observable copy:
+
+```javascript
+@observer
+class Alerter extends React.Component<{ counter: number }> {
+    @observable observableCounter: number
+    reactionDisposer
+
+    constructor(props) {
+        this.observableCounter = counter
+
+        // set up a reaction, by observing the observable,
+        // rather than the prop which is non-reactive:
+        this.reactionDisposer = autorun(() => {
```

Comment:
Move to `componenDidMount`, otherwise this could leak (react can create instance and throw it away without calling any lifecycle methods). For inspiration see tests:
`Observable prop workaround`
`Observable props/state/context workaround`

Reply:
Fixed